src/test/java/org/jenkinsci/test/acceptance/recorder/TestRecorderRuleTest.java

```diff
@@ -28,11 +32,15 @@ public void shouldNotRecordSuccessTestExecutionByDefault() {
 
         //Clean the field
         outputFile.delete();
+
+        restoreRecorderOption(oldMode);
```

Comment:
As currently written, it violates test isolation in case some of the tests fails and the temporary property value will leak to unrelated tests. Both `finally` and `@After` (configuring property from test, resetting old value stored in a field in `@After`) approach will give you the flexibility you are looking for.